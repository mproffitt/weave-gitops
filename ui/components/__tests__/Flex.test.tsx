import "jest-styled-components";
import { render } from "@testing-library/react";
import React from "react";
import Flex from "../Flex";

describe("Flex", () => {
  describe("snapshots", () => {
    it("wide", () => {
      const tree = render(<Flex wide>My Text</Flex>).asFragment().firstChild;
      expect(tree).toMatchSnapshot();
    });
    it("center", () => {
      const tree = render(<Flex center>My Text</Flex>).asFragment().firstChild;
      expect(tree).toMatchSnapshot();
    });
    it("align", () => {
      const tree = render(<Flex align>My Text</Flex>).asFragment().firstChild;
      expect(tree).toMatchSnapshot();
    });
    it("wide center align", () => {
      const tree = render(
        <Flex wide center align>
          My Text
        </Flex>,
      ).asFragment().firstChild;
      expect(tree).toMatchSnapshot();
    });
  });
});
