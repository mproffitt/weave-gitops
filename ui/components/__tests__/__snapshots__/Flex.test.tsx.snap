// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex snapshots align 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
}

<div
    class="c0"
  >
    My Text
  </div>
</DocumentFragment>
`;

exports[`Flex snapshots center 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: start;
  justify-content: center;
}

<div
    class="c0"
  >
    My Text
  </div>
</DocumentFragment>
`;

exports[`Flex snapshots wide 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: start;
  width: 100%;
}

<div
    class="c0"
  >
    My Text
  </div>
</DocumentFragment>
`;

exports[`Flex snapshots wide center align 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  justify-content: center;
}

<div
    class="c0"
  >
    My Text
  </div>
</DocumentFragment>
`;
